### PR TITLE
Switch from using choco to winget in rnw-dep script

### PR DESCRIPTION
**Pull request by acoates-ms**
Currently the rnw-dependencies script installs choco to install dependencies.
Current versions of windows should have winget available which can be used to install applications without requiring another package manager.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12070)

### CODE REVIEW

**Comment by chrisglein** [comment]
Fixes https://github.com/microsoft/react-native-windows/issues/11568

**Comment by chrisglein** [comment]
We should update https://microsoft.github.io/react-native-windows/docs/rnw-dependencies to remove Chocolatey as the "recommend" option for manual install. Separate repo, separate PR, but worth mentioning if we update the script.

File: vnext/Scripts/rnw-dependencies.ps1 [hunk 1]
```diff
@@ -473,12 +467,11 @@ $requirements = @(
         Tags = @('rnwDev');
         Valid = { ( cmd "/c assoc .binlog 2>nul" ) -ne $null; }
         Install = {
-            ChocoInstall msbuild-structured-log-viewer;
+            WinGetInstall KirillOsenkov.MSBuildStructuredLogViewer;
             $slv = gci ${env:LocalAppData}\MSBuildStructuredLogViewer\StructuredLogViewer.exe -Recurse | select FullName | Sort-Object -Property FullName -Descending | Select-Object -First 1
```
Comment (jonthysell) [inline vnext/Scripts/rnw-dependencies.ps1:471]: Do we still need this with the WinGet install?

File: vnext/Scripts/rnw-dependencies.ps1 [hunk 2]
```diff
@@ -515,63 +507,20 @@ $requirements = @(
...
-        Install = { ChocoInstall dotnet-$dotnetver-sdk };
-        InstallsWithChoco = $true;
+        Install = { WinGetInstall Microsoft.DotNet.SDK.$wingetDotNetVer };
         HasVerboseOutput = $true;
     }
 );
 
-$ChocoCmd = ""; # This should get set when EnsureChocoForInstall is called
-
-function EnsureChocoForInstall {
```
Comment (jonthysell) [inline vnext/Scripts/rnw-dependencies.ps1:526]: Do we have a solid min version of Windows 10 that includes winget by default? We currently specify 10.0.16299.0 as the min version to dev RNW.

Comment (acoates-ms) [reply vnext/Scripts/rnw-dependencies.ps1:526]: 👍 I'll look at adding some verification that winget is available and provide feedback if its not.

Comment (jonthysell) [reply vnext/Scripts/rnw-dependencies.ps1:526]: Ensure that feedback includes enough instructions for the user to both get winget installed and run it at least once to get past the license agreement

**Review by jonthysell — APPROVED** [review]

**Review by chiaramooney — APPROVED** [review]

**Pull request merged**